sql/src/test/java/io/crate/integrationtests/SubSelectGroupByIntegrationTest.java

```diff
@@ -66,20 +68,51 @@ public void testSelectWithWhereClause() throws Exception {
     @Test
     public void testDistributedSelectWithWhereClause() throws Exception {
         execute(
-            "select count(x) from (select x from t group by x limit 1) as tt " +
+            "select count(x) from (select x from t1 group by x limit 1) as tt " +
             "group by x " +
             "limit 1"
         );
         assertThat(TestingHelpers.printedTable(response.rows()), is("1\n"));
     }
 
     @Test
+    public void testAggregationWithGroupByAndOrderBy() throws Exception {
+        execute(
+            "select max(id), country from (select * from t3 order by 2 limit 1) as tt " +
+            "group by country " +
+            "order by 2"
+        );
+        assertThat(TestingHelpers.printedTable(response.rows()), is("1| 3\n"));
+    }
+
+    @Test
+    public void testAggregationWithJoinInSubselect() throws Exception {
+        execute(
+            "select max(country), count from (select count(*) as count, country from t3 group by country order by 2) as t " +
+            "group by count " +
+            "order by 1 " +
+            "limit 100 "
+        );
+        assertThat(TestingHelpers.printedTable(response.rows()), is("5| 1\n"));
+    }
+
+    @Test
+    public void testFunctionAndAggregationWithJoinInSubselect() throws Exception {
```

Comment:
I would move this to SubQueryPlannerTest as a "Unit test" where you can assert the GroupProjection's keys & values.
@mfussenegger do you agree?